src/common/transformations/src/transformations/common_optimizations/constants_reduce.cpp

```diff
@@ -0,0 +1,119 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/constants_reduce.hpp"
+
+#include "itt.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/util/log.hpp"
+
+#define LARGE_TENSOR_BYTE_SIZE 64
+
+namespace ov::pass {
+
+using BlobCacheKey = std::shared_ptr<ov::Node>;
+
+struct KeyHash {
+    std::size_t operator()(const BlobCacheKey& key) const {
+        std::size_t hash = 0;
+
+        auto node = ov::as_type_ptr<op::v0::Constant>(key);
+
+        auto type = node->get_output_element_type(0);
+        auto shape = node->get_shape();
+        std::size_t size = node->get_byte_size();
+        const char* data = node->get_data_ptr<char>();
+
+        for (auto dim : shape) {
+            hash ^= std::hash<size_t>{}(dim);
+        }
+
+        for (std::size_t i = 0; i < size; i++) {
+            hash ^= ((hash << 5) + hash) + data[i];
+        }
+
+        hash ^= type.hash();
+        hash ^= size;
+
+        return hash;
+    }
+};
+
+struct KeyEqual {
+    bool operator()(const BlobCacheKey& lhs, const BlobCacheKey& rhs) const {
+        auto lhs_node = ov::as_type_ptr<op::v0::Constant>(lhs);
+        auto rhs_node = ov::as_type_ptr<op::v0::Constant>(rhs);
+
+        auto lhs_type = lhs_node->get_output_element_type(0);
```

Comment:
This part of code looks similar to [this function](https://github.com/openvinotoolkit/openvino/blob/48b6c68f117b0a2d69a3d921d0a15355caf82f44/src/core/src/bound_evaluate.cpp#L38)

Consider expert these function to tensor_util.hpp (part of dev API ) and re-use it. The Constant node can provide tensor view